packages/Symfony/src/Rector/Class_/MakeCommandLazyRector.php

```diff
@@ -0,0 +1,129 @@
+ */
+final class MakeCommandLazyRector extends AbstractRector
+{
+    /**
+     * @var string
+     */
+    private const COMMAND_CLASS = 'Symfony\Component\Console\Command\Command';
+
+    /**
+     * @var CallableNodeTraverser
+     */
+    private $callableNodeTraverser;
+
+    public function __construct(CallableNodeTraverser $callableNodeTraverser)
+    {
+        $this->callableNodeTraverser = $callableNodeTraverser;
+    }
+
+    public function getDefinition(): RectorDefinition
+    {
+        return new RectorDefinition('Make Symfony commands lazy', [
+            new CodeSample(
+                <<<'CODE_SAMPLE'
+use Symfony\Component\Console\Command\Command
+
+class SunshineCommand extends Command
+{
+    public function configure()
+    {
+        $this->setName('sunshine');
+    }
+}
+CODE_SAMPLE
+                ,
+                <<<'CODE_SAMPLE'
+use Symfony\Component\Console\Command\Command
+
+class SunshineCommand extends Command
+{
+    protected static $defaultName = 'sunshine';
+    public function configure()
+    {
+    }
+}
+CODE_SAMPLE
+            ),
+        ]);
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getNodeTypes(): array
+    {
+        return [Class_::class];
+    }
+
+    /**
+     * @param Class_ $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        if (! $this->isType($node, self::COMMAND_CLASS)) {
+            return null;
+        }
+
+        $commandName = $this->resolveCommandNameAndRemove($node);
+        if ($commandName === null) {
+            return null;
+        }
+
+        $defaultNameProperty = $this->createDefaultNameProperty($commandName);
+        $node->stmts = array_merge([$defaultNameProperty], (array) $node->stmts);
+
+        return $node;
+    }
+
+    private function createDefaultNameProperty(string $commandName): Node\Stmt\Property
+    {
+        return $this->builderFactory->property('defaultName')
+            ->makeProtected()
+            ->makeStatic()
+            ->setDefault($commandName)
+            ->getNode();
+    }
+
+    private function resolveCommandNameAndRemove(Class_ $class): ?string
+    {
+        $commandName = null;
+        $this->callableNodeTraverser->traverseNodesWithCallable((array) $class->stmts, function (Node $node) use (
+            &$commandName
+        ) {
+            if ($node instanceof Node\Expr\MethodCall) {
+                if (! $this->isName($node, 'setName')) {
+                    return null;
+                }
+
+                $commandName = $this->getValue($node->args[0]->value);
+
+                $this->removeNode($node);
+            }
+
+            if ($node instanceof Node\Expr\StaticCall) {
+                if (! $this->isName($node, '__construct')) {
+                    return null;
+                }
+
+                $commandName = $this->getValue($node->args[0]->value);
+
+                $this->removeNode($node);
```

Comment:
Call to parent should not be removed cause, it will have side effects.

Reply:
So what should happen?

Reply:
You should leave empty call to parent:
```
parent::_construct();
```

Reply:
If you have in mind simple: why leave empty parent call. Answer is that by having commands as services you inject those sevices in constructor  ;)